vehicle/mkvi/software/throttle/throttle.c

```diff
@@ -0,0 +1,322 @@
+                throttle_debug.throttle_brake_implaus = true;
+                return true;
+            }
+        } else {
+            // no implausibility prev occurred then still no implausibility
+            throttle.throttle_state = THROTTLE_RUN;
+            throttle_debug.throttle_brake_implaus = false;
+            return false;
+        }
+    } else {
+        // brakes are not pressed
+        if (brake_implausibility_occurred) {
+            // but implausibility prev occurred
+            if (pos_min <= APPS_BRAKE_IMPLAUSIBILITY_THRESHOLD_LOW) {
+                // however pedal is <= 5% travel, no implausibility
+                brake_implausibility_occurred = false;
+                throttle_debug.throttle_brake_implaus = false;
+                throttle.throttle_state = THROTTLE_RUN;
+                return false;
+            } else {
+                // else pedal travel still >5%
+                throttle.throttle_state = THROTTLE_BRAKE_PRESSED;
+                throttle_debug.throttle_brake_implaus = true;
+                return true;
+            }
+        } else {
+            // brake not pressed, no prev implausibility
+            throttle.throttle_state = THROTTLE_RUN;
+            throttle_debug.throttle_brake_implaus = false;
+            return false;
+        }
+    }
+}
+
+int main(void) {
+    sei();
+
+    can_init_throttle();
+    adc_init();
+
+    timer_init(&timer0_cfg);
+    timer_init(&timer1_cfg);
+
+    updater_init(BTLDR_ID, 5);
+    gpio_set_mode(SS_IS, INPUT);
+    gpio_enable_interrupt(SS_IS);
+
+    throttle.throttle_state = THROTTLE_IDLE;
+
+    pcint0_callback();
+
+    /*
+     * This feature is added so that the inverter cannot be accidentally enabled
+     * when first powered up. This feature requires that before sending out an
+     * Inverter Enable command, the user must send out an Inverter Disable
+     * command. Once the inverter sees a Disable command, the lockout is removed
+     * and controller can receive the Inverter Enable command
+     */
+    can_send_m192_command_message();
+
+    m192_command_message.direction_command = MOTOR_CLOCKWISE;
+
+    can_receive_brakelight();
+    can_receive_dashboard();
+
+    while (true) {
+        updater_loop();
+        if (run_1ms) {
+            run_1ms = false;
+
+            if (can_poll_receive_brakelight() == 0) {
+                // checking brake status??
+                throttle_state.brake_pressed = brakelight.brake_gate;
+                can_receive_brakelight();
+            }
+
+            if (can_poll_receive_dashboard() == 0) {
+                // check RTD status
+                can_receive_dashboard();
+                bool ready_to_drive = dashboard.ready_to_drive;
+                if (ready_to_drive) {
+                    m192_command_message.inverter_enable = true;
+                } else {
+                    m192_command_message.inverter_enable = false;
+                }
+            }
```

Comment:
nit: This could be simplified to just assigning the ready to drive status to the inverter_enable status instead of creating a local variable to store it